src/Confluent.Kafka/Uuid.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Derived from: rdkafka-dotnet, licensed under the 2-clause BSD License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using Confluent.Kafka.Impl;
+using static Confluent.Kafka.Internal.Util.Marshal;
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents a UUID
+    /// </summary>
+    public class Uuid
+    {
+        /// <summary>
+        ///     Initializes a new instance of the Uuid class
+        ///     with the specified Most and Least significant bits.
+        /// </summary>
+        /// <param name="mostSignificantBits">
+        ///     Most significant 64 bits of the 128 bits UUID.
+        /// </param>
+        /// <param name="leastSignificantBits">
+        ///     Least significant 64 bits of the 128 bits UUID.
+        /// </param>
+        public Uuid(long mostSignificantBits, long leastSignificantBits)
+        {
+            MostSignificantBits = mostSignificantBits;
+            LeasttSignificantBits = leastSignificantBits;
+            IntPtr cUuid = Librdkafka.uuid_new(mostSignificantBits, leastSignificantBits);
+            IntPtr cBase64str = Librdkafka.uuid_base64str(cUuid);
```

Comment:
We should add null checks here, and do cleanup if null occurs.

Reply:
With latest changes conversion to string is always available even without OpenSSL

Reply:
With latest changes conversion to string is always available even without OpenSSL